seed/src/lib.rs

```diff
@@ -270,30 +270,50 @@ impl Node {
     ) -> Result<(), Error> {
         let peer_id = peer_info.peer_id;
         let url = urn.clone().into_rad_url(peer_id);
+        let proejct_urn = RadUrn {
+            path: uri::Path::new(),
+            ..urn.clone()
+        };
         let port = peer_info.advertised_info.listen_port;
         let addr_hints = peer_info
             .seen_addrs
             .iter()
             .map(|a: &std::net::IpAddr| (*a, port).into())
             .collect::<Vec<_>>();
 
-        let result = {
+        // FIXME(xla): There should be a saner way to test.
+        let exists = {
+            let urn = proejct_urn.clone();
+            api.with_storage(move |storage| storage.has_urn(&urn))
+                .await??
+        };
+
```

Comment:
Shouldn't we try to clone, and if that fails, try to fetch? I'm wondering if we get in a situation where the pre-condition changes in between calls to `with_storage`. Eg. it exists, then it's deleted, then we try to fetch. Or are we always assuming there is only one writer to the monorepo on the device where this runs?

Reply:
What would delete the project? And changing the order won't help if the precondition changes in between.

Reply:
> What would delete the project?

That's my question, but for eg. some other radicle process running, eg. upstream etc.

> And changing the order won't help if the precondition changes in between.

That's not what I'm suggesting, I'm suggesting getting rid of the `exist` check so that there's no race condition possible.

Reply:
> That's not what I'm suggesting, I'm suggesting getting rid of the `exist` check so that there's no race condition possible.

The race condition is still possible, if we fetch after a clone has failed and the project goes away in between the clone and fetch, it still produces an error.

Reply:
> That's my question, but for eg. some other radicle process running, eg. upstream etc.

A seed and an upstream would not share the same state.